docs/walkthrough/walkthrough-oauth2-proxy.md

```diff
@@ -0,0 +1,168 @@
+
+To install the chart, simply set the `Client ID` and `Client Secret` you just obtained from Github and run the command below:
+
+```bash
+# CLIENT_ID and CLIENT_SECRET are the Client ID and Client Secret obtained
+# when creating the Github OAuth application in the previous step
+CLIENT_ID=__THE_CLIENT_ID_OF_YOUR_GITHUB_OAUTH_APP__
+CLIENT_SECRET=__THE_CLIENT_SECRET_OF_YOUR_GITHUB_OAUTH_APP__
+
+cat <<EOF | helm upgrade --install --wait --create-namespace --namespace tools oauth2-proxy stable/oauth2-proxy --values -
+config:
+  clientID: $CLIENT_ID
+  clientSecret: $CLIENT_SECRET
+
+extraArgs:
+  provider: github
+  whitelist-domain: .127.0.0.1.nip.io
+  cookie-domain: .127.0.0.1.nip.io
+  redirect-url: http://auth.127.0.0.1.nip.io/oauth2/callback
+  cookie-secure: 'false'
+
+ingress:
+  enabled: true
+  path: /
+  hosts:
+    - auth.127.0.0.1.nip.io
+EOF
+```
+
+This will install `oauth2-proxy` in your cluster in the `tools` namespace.
+
+Some parameters in this configuration deserve a bit of explanations:
+- `config.ClientID` and `config.clientSecret` are obtained when creating the Github OAuth application
+- `extraArgs.provider` tells `oauth2-proxy` to use Github
+- `extraArgs.redirect-url` is needed for HTTP callbacks, `oauth2-proxy` will default to HTTPS if not set
+- `extraArgs.whitelist-domain` allows `oauth2-proxy` to authenticate request for all `.127.0.0.1.nip.io` subdomains
+- `extraArgs.cookie-secure` disables CSRF as we are using HTTP (not HTTPS)
+- `extraArgs.cookie-domain` is used to set the cookie for all `.127.0.0.1.nip.io` subdomains. As `oauth2-proxy` is deployed centrally, it will be exposed at `http://auth.127.0.0.1.nip.io` while the dashboard will be exposed at `http://tekton-dashboard.127.0.0.1.nip.io`. The cookie needs to be available to both urls
+
+Although not used in this walkthrough, Github provider supports restricting logins to members of an organisation, members of a team, or collaborator of a repository for more security.
+
+See supported [configuration parameters](https://oauth2-proxy.github.io/oauth2-proxy/configuration) for the full list of confiuration options.
+
+`oauth2-proxy` also supports other providers, see the list of [supported providers](https://oauth2-proxy.github.io/oauth2-proxy/auth-configuration) to setup a different provider than Github.
+
+## Setting up Dashboard Ingress rule for authentication
+
+With `oauth2-proxy` deployed, you will now secure the Tekton Dashboard `Ingress`.
+
+To do so, you need to add [annotations on the ingress](https://github.com/kubernetes/ingress-nginx/blob/master/docs/user-guide/nginx-configuration/annotations.md#external-authentication) to let `NGINX ingress controller` know that the ingress requires authentication and how to authenticate.
+
+Two annotations are used here:
+- `nginx.ingress.kubernetes.io/auth-url` indicates that the ingress uses an external service that provides authentication. This will point to `oauth2-proxy` service
```

Comment:
```suggestion
- `nginx.ingress.kubernetes.io/auth-url` indicates that the ingress uses an external service that provides authentication. This will point to the `oauth2-proxy` service
```